### ISSUE TITLE
Asynchronous error

### ISSUE THREAD

**Issue by berkobob** [mongo-dart/mongo_dart#213]
My code:

```
import 'dart:io';

import 'package:mongo_dart/mongo_dart.dart';

class Database {
  Db db;

  void start() async {
    db = await Db.create(Platform.environment['DBURL']);
    await db.open(secure: true);
    print('Database open');
  }

  Future<dynamic> get(String collectionName) async {
    final collection = db.collection(collectionName);
    return await collection.find().toList();
  }
}
```

When I run get I see the following error:

> ERROR - 2021-02-09 13:51:15.748154
> Asynchronous error
> MongoDB ConnectionException: connection closed: The socket connection has been reset by peer.
> Possible causes:
> - Trying to connect to an ssl/tls encrypted database without specifiyng
>   either the query parm tls=true or the secure=true parameter in db.open()
> - Others
> 

What's going on? Thank you.

**Comment by giorgiofran** [comment]
Which version of Mongo_dart are you using?
If you try from the Mongo shell, are you able to connect?
If it is an Atlas Db, have you whitelisted your IP?

**Comment by berkobob** [comment]
>   mongo_dart: ^0.4.4

Yes, I can connect via the shell without problems
The Atlas Db is currently open during development

**Comment by giorgiofran** [comment]
Platform.environment['DBURL'] is a correct mongodb+srv:// url? Note that i should be different form the one to be used in shell

**Comment by giorgiofran** [comment]
Did you try with the last (0.5.0-dev) version?

**Comment by berkobob** [comment]
Yes, it's a valid URL but no, I haven't tried 0.5.0-dev. Is it stable?

**Comment by giorgiofran** [comment]
It is quite stable, API are still subject to change.

**Comment by berkobob** [comment]
That worked. Thank you very much.

**Comment by giorgiofran** [comment]
I recently discovered that there is a problem with authentication in 0.4.4 (solved in 0.5.0). 
In order to connect an user must have (in 0.4.4) the authorization of running the serverStatus command. This is not the case for all users.
So, maybe that the problem was due to user authentication.

**Comment by giorgiofran** [comment]
I'm closing this issue. Feel free to ask for any other question
